packages/xeus/src/worker.ts

```diff
@@ -173,40 +192,271 @@ export abstract class EmpackedXeusRemoteKernel extends XeusRemoteKernelBase {
     });
   }
 
+  protected deleteFromInstalledPackages(
+    pkgName: string,
+    installed: ISolvedPackages
+  ) {
+    const newInstalledPkgs: ISolvedPackages = {};
+    Object.keys(installed).map((filename: string) => {
+      const installedPkg = installed[filename];
+      if (pkgName !== installedPkg.name) {
+        newInstalledPkgs[filename] = installedPkg;
+      }
+    });
+    return newInstalledPkgs;
+  }
+
+  protected getPackageName(specs: string) {
+    const nameMatch = specs.match(/^([a-zA-Z0-9_-]+)/);
+
+    if (!nameMatch) {
+      return null;
+    }
+
+    const packageName = nameMatch[1];
+    return packageName;
+  }
+
```

Comment:
Could you please move this as a free function in mambajs-core so we can reuse it in multiple places?

I'd rename it `packageNameFromSpec(spec: string)`